host/xtest/xtest_1000.c

```diff
@@ -440,9 +441,19 @@ static FILE *open_ta_file(const TEEC_UUID *uuid, const char *mode,
 			  bool for_write)
 {
 	char buf[PATH_MAX];
+	FILE *ret;
 
 	uuid_to_full_name(buf, sizeof(buf), uuid, for_write);
-	return fopen(buf, mode);
+	ret = fopen(buf, mode);
+	if (ret && for_write)
+		if (chmod(buf, 0666)) {
```

Comment:
That was my first thought too but looking it up, it seems like `fopen()` can only do the `r, w, r+`, etc stuffs?